lib/core/src/refund/chain.rs

```diff
@@ -0,0 +1,273 @@
+    bitcoin_chain_service: Arc<Mutex<dyn BitcoinChainService>>,
+}
+
+impl ChainRefundHandler<BoltzSwapper> {
+    pub(crate) fn new(
+        config: Config,
+        swapper: Arc<BoltzSwapper>,
+        liquid_chain_service: Arc<Mutex<dyn LiquidChainService>>,
+        bitcoin_chain_service: Arc<Mutex<dyn BitcoinChainService>>,
+    ) -> Self {
+        Self {
+            config,
+            swapper: Some(swapper),
+            liquid_chain_service,
+            bitcoin_chain_service,
+        }
+    }
+
+    fn boltz_url(&self) -> &str {
+        self.swapper
+            .as_ref()
+            .map(|swapper| swapper.boltz_url())
+            .unwrap_or(match self.config.network {
+                LiquidNetwork::Mainnet => BOLTZ_MAINNET_URL_V2,
+                LiquidNetwork::Testnet => BOLTZ_TESTNET_URL_V2,
+            })
+    }
+
+    fn bitcoin_electrum_config(&self) -> ElectrumConfig {
+        self.swapper
+            .as_ref()
+            .map(|swapper| swapper.bitcoin_electrum_config().clone())
+            .unwrap_or(ElectrumConfig::new(
+                self.config.network.as_bitcoin_chain(),
+                &self.config.bitcoin_electrum_url,
+                true,
+                true,
+                100,
+            ))
+    }
+
+    pub(crate) async fn refund_incoming_swap(
+        &self,
+        swap: &ChainSwap,
+        refund_address: &str,
+        is_cooperative: bool,
+        fee_rate_msats_per_vbyte: u32,
+    ) -> Result<String, PaymentError> {
+        if swap.direction != Direction::Incoming {
+            return Err(PaymentError::Generic {
+                err: "Cannot refund an outgoing swap via this method".to_string(),
+            });
+        }
+
+        info!(
+            "Initiating refund for incoming Chain Swap {}, is_cooperative: {is_cooperative}",
+            swap.id
+        );
+
+        let SwapScriptV2::Bitcoin(swap_script) = swap.get_lockup_swap_script()? else {
+            return Err(PaymentError::Generic {
+                err: "Unexpected swap script type found".to_string(),
+            });
+        };
+        let refund_keypair = swap.get_refund_keypair()?;
+        let cooperative = self.swapper.as_ref().and_then(|swapper| {
+            common::get_cooperative(&swap.id, is_cooperative, swapper.client())
+        });
+
+        let bitcoin_electrum_config = self.bitcoin_electrum_config();
+
+        let refund_tx = BtcSwapTx::new_refund(
+            swap_script.clone(),
+            &refund_address.into(),
+            &bitcoin_electrum_config,
+            self.boltz_url().to_string(),
+            swap.id.clone(),
+        )?;
+        let broadcast_fees_sat = common::estimate_btc_refund_fees_sat(
+            &refund_tx,
+            &refund_keypair,
+            &Preimage::from_str(&swap.preimage)?,
+            fee_rate_msats_per_vbyte as f64 / 1000.0,
+        )?;
+        let signed_tx = refund_tx.sign_refund(&refund_keypair, broadcast_fees_sat, cooperative)?;
```

Comment:
The feedback here is also for other refund handlers:
The attempt to put the refund logic outside the swapper enforces us to expose the swapper internals (bolt client, bitcoin electrum config, etc...) and it makes the abstraction useless.
Since the cooperative refund needs the swapper interaction it makes sense to have it inside the swapper IMO.
As for the non cooperative refund I have not objection of having it outside the swapper (if it doesn't expose the internals) but if the cooperative will move there perhaps it would be more simple to have the non cooperative there as well.
The only thing IMO that should be outside the swapper is the broadcast itself.